File: server/Timelogger/Entities/Task.cs
namespace Timelogger.Entities
{
    public class Task
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int TimeSpent { get; set; }
        public int ProjectId { get; set; }
        public int? TaskTypeId { get; set; }
        public int? TaskCategoryId { get; set; }
    }
}

File: client/src/shared/types/Task.ts
import { Status } from "./Project";

export type TaskCategory = {
  id: number;
  name: string;
};

export type TaskType = {
  id: string;
  name: string;
  icon?: string;
};

export type TaskData = { categoryName?: string; typeName?: string };

export type Task = {
  id: number;
  title: string;
  timeSpent: string;
  projectId: number;
  userId: number;
  status: Status;
  statusName?: string;
  category?: TaskCategory;
  type?: TaskType;
} & TaskData;

export type TaskFormParams = {
  status: string;
  taskCategoryId?: string;
  taskTypeId?: string;
  timeSpent: string;
  title: string;
};
